packages/strapi-parts/src/Combobox/Combobox.js

```diff
@@ -176,42 +189,85 @@ export const Combobox = ({
+    inputRef.current.focus();
+
+    if (onClear) {
+      onClear();
+    }
+
+    setInputValue('');
+  };
+
+  const handleCaretClick = () => {
+    inputRef.current.focus();
+    updateMenuState(true);
+  };
+
   return (
     <Field hint={hint} error={error} id={generatedId}>
       <Stack size={label || hint || error ? 1 : 0}>
         {label && <FieldLabel id={labelId}>{label}</FieldLabel>}
         <MainRow ref={containerRef} $disabled={disabled} hasError={error}>
-          <Input
-            id={generatedId}
-            aria-activedescendant={activeId}
-            aria-autocomplete="list"
-            aria-controls={`${generatedId}-listbox`}
-            aria-expanded={`${open}`}
-            aria-haspopup="listbox"
-            aria-describedby={label ? labelId : undefined}
-            aria-disabled={disabled}
-            readOnly={disabled}
-            ref={inputRef}
-            role="combobox"
-            type="text"
-            value={inputValue}
-            onBlur={disabled ? undefined : onInputBlur}
-            onClick={disabled ? undefined : () => updateMenuState(true)}
-            onInput={disabled ? undefined : onInput}
-            onKeyDown={disabled ? undefined : onInputKeyDown}
-            placeholder={placeholder}
-            {...props}
-          />
+          {disabled ? (
+            <Input
+              id={generatedId}
+              aria-activedescendant={activeId}
+              aria-autocomplete="list"
+              aria-controls={`${generatedId}-listbox`}
+              aria-expanded={open}
+              aria-haspopup="listbox"
+              aria-labelledby={label ? labelId : undefined}
+              aria-disabled
+              readOnly
+              ref={inputRef}
+              role="combobox"
+              type="text"
+              value={inputValue}
+              placeholder={placeholder}
+              {...props}
+            />
+          ) : (
+            <Input
+              id={generatedId}
+              aria-activedescendant={activeId}
+              aria-autocomplete="list"
+              aria-controls={`${generatedId}-listbox`}
+              aria-expanded={open}
+              aria-haspopup="listbox"
+              aria-labelledby={label ? labelId : undefined}
+              ref={inputRef}
+              role="combobox"
+              type="text"
+              value={inputValue}
+              onBlur={onInputBlur}
+              onClick={() => updateMenuState(true)}
+              onInput={onInput}
+              onKeyDown={onInputKeyDown}
+              placeholder={placeholder}
+              {...props}
+            />
+          )}
           <Row>
+            {inputValue && (
+              <IconBox
+                id={`${generatedId}-clear`}
+                disabled={disabled}
+                paddingLeft={3}
+                aria-hidden
+                as="button"
+                onClick={handleClear}
+                tabIndex={-1}
```

Comment:
You should remove this one, we want the clear button to be keyboard accessible :)